src/test/regress/expected/dml_recursive.out

```diff
@@ -0,0 +1,317 @@
+		WHERE 
+			d1.tenant_id = second_distributed_table.tenant_id
+		AND
+			second_distributed_table.dept IN (3,4)
+			AND 
+			second_distributed_table.tenant_id IN 
+			(
+					SELECT s2.tenant_id
+					FROM second_distributed_table as s2
+					GROUP BY d1.tenant_id, s2.tenant_id
+			) 
+	) as baz
+	) as foo WHERE second_distributed_table.tenant_id = foo.tenant_id
+RETURNING *;
+DEBUG:  complex joins are only supported when all distributed tables are joined on their distribution columns with equal operator
+ERROR:  complex joins are only supported when all distributed tables are joined on their distribution columns with equal operator
+-- we don't support subquerues/CTEs inside VALUES
+INSERT INTO 
+	second_distributed_table (tenant_id, dept) 
+VALUES ('3', (WITH  vals AS (SELECT 3) select * from vals));
+DEBUG:  subqueries are not supported within INSERT queries
+HINT:  Try rewriting your queries with 'INSERT INTO ... SELECT' syntax.
+DEBUG:  generating subplan 18_1 for CTE vals: SELECT 3
+DEBUG:  Plan 18 query after replacing subqueries and CTEs: INSERT INTO recursive_dml_queries.second_distributed_table (tenant_id, dept) VALUES ('3'::text, (SELECT vals."?column?" FROM (SELECT intermediate_result."?column?" FROM read_intermediate_result('18_1'::text, 'binary'::citus_copy_format) intermediate_result("?column?" integer)) vals))
+DEBUG:  subqueries are not supported within INSERT queries
+HINT:  Try rewriting your queries with 'INSERT INTO ... SELECT' syntax.
+ERROR:  subqueries are not supported within INSERT queries
+HINT:  Try rewriting your queries with 'INSERT INTO ... SELECT' syntax.
+INSERT INTO 
+	second_distributed_table (tenant_id, dept) 
+VALUES ('3', (SELECT 3));
+DEBUG:  subqueries are not supported within INSERT queries
+HINT:  Try rewriting your queries with 'INSERT INTO ... SELECT' syntax.
+ERROR:  subqueries are not supported within INSERT queries
+HINT:  Try rewriting your queries with 'INSERT INTO ... SELECT' syntax.
+-- we error out of the CTE is not referenced
+-- by any part of the query (e.g., cte2 is not referenced)
+WITH cte_1 AS (
+    WITH cte_2 AS (
+        SELECT tenant_id as cte2_id 
+        FROM second_distributed_table 
+        WHERE dept >= 2
+    )
+    
+    UPDATE distributed_table 
+    SET dept = 10
+    RETURNING *
+)
+UPDATE distributed_table
+SET dept = 5
+FROM cte_1
+WHERE distributed_table.tenant_id < cte_1.tenant_id;
+DEBUG:  common table expressions are not supported in distributed modifications
+DEBUG:  generating subplan 20_1 for CTE cte_1: WITH cte_2 AS (SELECT second_distributed_table.tenant_id AS cte2_id FROM recursive_dml_queries.second_distributed_table WHERE (second_distributed_table.dept >= 2)) UPDATE recursive_dml_queries.distributed_table SET dept = 10 RETURNING tenant_id, dept, info
+DEBUG:  common table expressions are not supported in distributed modifications
+ERROR:  common table expressions are not supported in distributed modifications
```

Comment:
In general `common table expressions are not supported in distributed modifications` seems not a very accurate error message anymore. 

As far as I can see, it could only happen when a `SELECT` cte is not referenced at all.

Shall we update the error message accordingly? 

(I'm OK not fixing it as well since it is a very minor thing)

Reply:
Fixed the underlying issue, so these queries work now.

 We may want to clean up debug messages at some point.